### PR TITLE
Feat/dnsdid did revocation

### PR DESCRIPTION
**Pull request by cavacado**
context: https://www.pivotaltracker.com/story/show/176199971
this PR does the following:
1. adds a new generating function in `generate.v3` scripts for these two types of files (1. signed and **not** revoked from a document store; 2. signed and **revoked** from a document store) for the file variants DID, and DNS-DID:
2. adds the newly generated files for v2 and v3
3. modifies `didSignedDocumentStatus` file to introduce support for revocation for v2 and v3 DID / DNS-DID documents
4. adds/updates unit test for didSignedDocumentStatus
5. adds/updates integration tests for v2 and v3 files.
6. and some misc refactoring here and there

### CODE REVIEW

File: scripts/generate.v3.ts [hunk 1]
```diff
@@ -35,6 +35,80 @@ const ethereumDocumentConfig = {
...
+      revocation: {
+        type: v3.RevocationType.RevocationStore,
+        location: ethereumDocumentConfig.documentStore,
+      },
+    },
+  },
+};
+
+const generateRevocationStore = async ({ logMessage, document, outLocations }: GenerateRevocationStoreArgs) => {
+  info(`${logMessage}`);
```
Comment (yehjxraymond) [inline scripts/generate.v3.ts:76]: Wondering why you parameterized this. There is a coupling in between the two function in which the caller knows that the log message will be logged as is at the top.

**Review by yehjxraymond — CHANGES_REQUESTED** [review]
Need to check intermediate hashes when checking for revocation of a document.

File: src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.ts [hunk 2]
```diff
@@ -54,8 +60,32 @@ const verifyV2 = async (document: SignedWrappedDocument<v2.OpenAttestationDocume
...
-  const revokedOnAny = !revocation.every((r) => r?.type === "NONE");
+  const revocationStatusCallback = (revocationItem: v2.Revocation) => {
+    switch (revocationItem.type) {
+      case v2.RevocationType.RevocationStore:
+        return isRevokedOnDocumentStore({
+          documentStore: revocationItem.location ?? "",
+          merkleRoot,
+          provider: options.provider,
+          targetHash,
+        });
```
[the remark this answers is not in the document]
Comment (cavacado) [reply src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.ts:71]: i have no idea why I left this out, must have copied something wrongly. 😓

File: src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.ts [hunk 3]
```diff
@@ -54,8 +60,32 @@ const verifyV2 = async (document: SignedWrappedDocument<v2.OpenAttestationDocume
       "MISSING_REVOCATION"
```
[the remark this answers is not in the document]
Comment (cavacado) [reply src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.ts:60]: sure

File: src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.ts [hunk 4]
```diff
@@ -143,7 +181,35 @@ const verifyV3 = async (document: SignedWrappedDocument<v3.OpenAttestationDocume
...
+          "revocation type not found for an issuer",
+          OpenAttestationDidSignedDocumentStatusCode.UNRECOGNIZED_REVOCATION_TYPE,
+          "UNRECOGNIZED_REVOCATION_TYPE"
+        );
+    }
+  };
+
+  const revocationStatus = await getRevocationStatus(
+    metaData.proof.revocation.type,
+    metaData.proof.revocation.location ?? ""
```
[the remark this answers is not in the document]
Comment (cavacado) [reply src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.ts:208]: yup will do this.

File: src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.test.ts [hunk 5]
```diff
@@ -156,25 +258,19 @@ describe("verify", () => {
         }
       `);
     });
-    it("should fail when revocation is not set to NONE (for now)", async () => {
+    it("should throw an unrecognized revocation type error when revocation is not set to NONE || REVOCATION_STORE (for now)", async () => {
```
[the remark this answers is not in the document]
Comment (cavacado) [reply src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.test.ts:261]: 👌

File: src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.test.ts [hunk 6]
```diff
@@ -326,7 +563,7 @@ describe("verify", () => {
       `);
     });
 
-    it("should fail when revocation is not set to NONE (for now)", async () => {
+    it("should throw an unrecognized revocation type error when revocation is not set to NONE || REVOCATION_STORE (for now)", async () => {
```
[the remark this answers is not in the document]
Comment (cavacado) [reply src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.test.ts:566]: 👌

**Comment by cavacado** [comment]
@yehjxraymond have done the requested changes and added a check for a null revocation location (else throw) if `REVOCATION_STORE` is used

File: src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.ts [hunk 7]
```diff
@@ -182,16 +189,23 @@ const verifyV3 = async (
 
   const issuedOnAll = verificationResult.verified;
 
-  const getRevocationStatus = async (docType: v3.RevocationType, location: string) => {
+  const getRevocationStatus = async (docType: v3.RevocationType, location: string | undefined) => {
```
Comment (yehjxraymond) [inline src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.ts:192]:
```suggestion
  const getRevocationStatus = async (docType: v3.RevocationType, location?: string) => {
```

Hmm i think we use `?` most of the time right?

**Review by yehjxraymond — APPROVED** [review]
lgtm

**Pull request merged**